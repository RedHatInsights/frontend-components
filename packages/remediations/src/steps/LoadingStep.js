import React, { PureComponent } from 'react';
import propTypes from 'prop-types';

import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/components/cjs/Skeleton';
import { Stack, StackItem } from '@patternfly/react-core';

import './LoadingStep.scss';

export default class LoadingStep extends PureComponent {

    componentDidMount() {
        this.props.onValidChange(false);
    }

    componentWillUnmount () {
        this.props.onValidChange(true);
    }

    render () {
        return (
            <Stack gutter='sm' className='ins-c-remediations-loading-step'>
                <StackItem><h1><Skeleton size={ SkeletonSize.sm } /></h1></StackItem>
                <StackItem>
                    <Skeleton size={ SkeletonSize.lg } className='ins-c-remediations-loading-step-main' />
                </StackItem>
                <StackItem>
                    <Stack gutter='sm'>
                        <StackItem>
                            <Skeleton size={ SkeletonSize.sm } />
                        </StackItem>
                        <StackItem>
                            <Skeleton size={ SkeletonSize.xs } />
                        </StackItem>
                    </Stack>
                </StackItem>
            </Stack>
        );
    }
}

LoadingStep.propTypes = {
    onValidChange: propTypes.func.isRequired
};

