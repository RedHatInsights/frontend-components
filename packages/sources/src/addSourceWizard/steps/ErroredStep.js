import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { ErrorCircleOIcon } from '@patternfly/react-icons';

const ErroredStep = ({ onClose, onRetry }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ ErrorCircleOIcon } color="var(--pf-global--danger-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration unsuccessful
        </Title>
        <EmptyStateBody>
      Something went wrong. Your source has not been successfully added.
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <Button variant="link" onClick={ onRetry }>Retry</Button>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

ErroredStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    onRetry: PropTypes.func.isRequired
};

export default ErroredStep;
