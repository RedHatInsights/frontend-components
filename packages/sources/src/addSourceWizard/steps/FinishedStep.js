import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';

const FinishedStep = ({ onClose, successfulMessage }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ CheckCircleIcon } color="var(--pf-global--success-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration successful
        </Title>
        <EmptyStateBody>
            { successfulMessage }
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <a href='/hybrid/settings/sources'>
                <Button variant="link">Take me to sources</Button>
            </a>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

FinishedStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    successfulMessage: PropTypes.node.isRequired
};

export default FinishedStep;
