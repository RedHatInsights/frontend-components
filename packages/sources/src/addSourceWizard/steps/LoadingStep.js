import React from 'react';
import PropTypes from 'prop-types';
import { EmptyState, EmptyStateVariant, EmptyStateBody, EmptyStateSecondaryActions, Button } from '@patternfly/react-core';
import { Spinner } from '@redhat-cloud-services/frontend-components';

const LoadingStep = ({ onClose, customText }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <div>
            <Spinner />
        </div>
        <EmptyStateBody>
            { customText }
        </EmptyStateBody>
        { onClose &&
        <EmptyStateSecondaryActions>
            <Button variant="link" onClick={ onClose }>Cancel</Button>
        </EmptyStateSecondaryActions> }
    </EmptyState>
</div>;

LoadingStep.propTypes = {
    onClose: PropTypes.func,
    customText: PropTypes.string
};

LoadingStep.defaultProps = {
    customText: 'Loading, please wait.'
};

export default LoadingStep;
