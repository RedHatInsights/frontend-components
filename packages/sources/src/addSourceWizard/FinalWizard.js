import React from 'react';
import PropTypes from 'prop-types';
import { Wizard } from '@patternfly/react-core';

import FinishedStep from './steps/FinishedStep';
import ErroredStep from './steps/ErroredStep';
import LoadingStep from './steps/LoadingStep';

const FinalWizard = ({ afterSubmit, afterError, isFinished, isErrored, onRetry, successfulMessage, hideSourcesButton, returnButtonTitle }) =>
    <Wizard
        isOpen={ true }
        onClose={ isFinished ? afterSubmit : afterError }
        title="Add a source"
        description="Connect an external source to Red Hat Cloud Services"
        steps={ [{
            name: 'Finish',
            component: isFinished ?
                <FinishedStep
                    onClose={ afterSubmit }
                    successfulMessage={ successfulMessage }
                    hideSourcesButton={ hideSourcesButton }
                    returnButtonTitle={ returnButtonTitle }
                /> :
                isErrored ?
                    <ErroredStep onRetry={ onRetry } onClose={ afterError } returnButtonTitle={ returnButtonTitle }/>
                    : <LoadingStep customText='Source is being created'/>,
            isFinishedStep: true
        }] } />;

FinalWizard.propTypes = {
    afterSubmit: PropTypes.func.isRequired,
    afterError: PropTypes.func.isRequired,
    onRetry: PropTypes.func.isRequired,
    isFinished: PropTypes.bool.isRequired,
    isErrored: PropTypes.bool.isRequired,
    successfulMessage: PropTypes.node.isRequired,
    hideSourcesButton: PropTypes.bool.isRequired,
    returnButtonTitle: PropTypes.node.isRequired
};

export default FinalWizard;
