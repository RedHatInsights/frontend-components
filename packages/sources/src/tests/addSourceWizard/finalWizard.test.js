
import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FinalWizard from '../../addSourceWizard/FinalWizard';
import FinishedStep from '../../addSourceWizard/steps/FinishedStep';
import LoadingStep from '../../addSourceWizard/steps/LoadingStep';
import ErroredStep from '../../addSourceWizard/steps/ErroredStep';

describe('Final wizard', () => {
    describe('Renders', () => {
        let initialProps;

        beforeEach(() => {
            initialProps = {
                afterSubmit: jest.fn(),
                afterError: jest.fn(),
                onRetry: jest.fn(),
                isFinished: false,
                isErrored: false,
                successfulMessage: 'Message',
                hideSourcesButton: false
            };
        });

        it('renders loading step correctly', () => {
            const wrapper = shallow(<FinalWizard { ...initialProps }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('contains loading step', () => {
            const wrapper = mount(<FinalWizard { ...initialProps }/>);
            expect(wrapper.find(LoadingStep).length).toBe(1);
        });

        it('renders finished step correctly', () => {
            const wrapper = shallow(<FinalWizard { ...initialProps } isFinished={ true }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('contains finished step', () => {
            const wrapper = mount(<FinalWizard { ...initialProps } isFinished={ true }/>);
            expect(wrapper.find(FinishedStep).length).toBe(1);
        });

        it('renders errored step correctly', () => {
            const wrapper = shallow(<FinalWizard { ...initialProps } isErrored={ true }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('contains errored step', () => {
            const wrapper = mount(<FinalWizard { ...initialProps } isErrored={ true }/>);
            expect(wrapper.find(ErroredStep).length).toBe(1);
        });
    });
});
