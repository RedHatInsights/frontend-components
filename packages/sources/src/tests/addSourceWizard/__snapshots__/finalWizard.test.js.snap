// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Final wizard Renders renders errored step correctly 1`] = `
<Wizard
  ariaLabelCloseButton="Close"
  ariaLabelNav="Steps"
  backButtonText="Back"
  cancelButtonText="Cancel"
  className=""
  description="Connect an external source to Red Hat Cloud Services"
  footer={null}
  hasBodyPadding={true}
  height={null}
  isCompactNav={false}
  isFullHeight={false}
  isFullWidth={false}
  isOpen={true}
  nextButtonText="Next"
  onBack={null}
  onClose={[MockFunction]}
  onGoToStep={null}
  onNext={null}
  startAtStep={1}
  steps={
    Array [
      Object {
        "component": <ErroredStep
          onClose={[MockFunction]}
          onRetry={[MockFunction]}
          returnButtonTitle="Go back to my application"
        />,
        "isFinishedStep": true,
        "name": "Finish",
      },
    ]
  }
  title="Add a source"
  width={null}
/>
`;

exports[`Final wizard Renders renders finished step correctly 1`] = `
<Wizard
  ariaLabelCloseButton="Close"
  ariaLabelNav="Steps"
  backButtonText="Back"
  cancelButtonText="Cancel"
  className=""
  description="Connect an external source to Red Hat Cloud Services"
  footer={null}
  hasBodyPadding={true}
  height={null}
  isCompactNav={false}
  isFullHeight={false}
  isFullWidth={false}
  isOpen={true}
  nextButtonText="Next"
  onBack={null}
  onClose={[MockFunction]}
  onGoToStep={null}
  onNext={null}
  startAtStep={1}
  steps={
    Array [
      Object {
        "component": <FinishedStep
          hideSourcesButton={false}
          onClose={[MockFunction]}
          returnButtonTitle="Go back to my application"
          successfulMessage="Message"
        />,
        "isFinishedStep": true,
        "name": "Finish",
      },
    ]
  }
  title="Add a source"
  width={null}
/>
`;

exports[`Final wizard Renders renders loading step correctly 1`] = `
<Wizard
  ariaLabelCloseButton="Close"
  ariaLabelNav="Steps"
  backButtonText="Back"
  cancelButtonText="Cancel"
  className=""
  description="Connect an external source to Red Hat Cloud Services"
  footer={null}
  hasBodyPadding={true}
  height={null}
  isCompactNav={false}
  isFullHeight={false}
  isFullWidth={false}
  isOpen={true}
  nextButtonText="Next"
  onBack={null}
  onClose={[MockFunction]}
  onGoToStep={null}
  onNext={null}
  startAtStep={1}
  steps={
    Array [
      Object {
        "component": <LoadingStep
          customText="Source is being created"
        />,
        "isFinishedStep": true,
        "name": "Finish",
      },
    ]
  }
  title="Add a source"
  width={null}
/>
`;
